src/mlpack/core/data/split_data.hpp

```diff
@@ -224,6 +229,50 @@ Split(const arma::Mat<T>& input,
                          std::move(testData));
 }
 
+template <class fieldType,
+          class = std::enable_if_t<
+              arma::is_Col<typename fieldType::object_type>::value ||
+              arma::is_Mat_only<typename fieldType::object_type>::value>>
+void Split(fieldType& input, 
+           arma::field<arma::vec>& inputLabel,
```

Comment:
@kartikdutt18 , could you (once again :D) , explain exactly what input and inputLabel are we expecting here?
If input is supposed to be image, then putting ```fieldType& input```  as images are stored as arma::vec. The current template allows ```input``` to have ```field<arma::vec> ``` or ```field<arma::mat>``` 

 Bounding boxes are passed around as ```mat / field<vec>``` I think ? So why allow ```field<mat>```

Reply:
The aim of the issue is to extend support for multiple / majority datatypes. The implementation in models repo is only for reference to see which functions would be needed for field type. It doesn't really matter what is inside the field type, if you get indices and create a new field type and equate the indices equal to indices in original dataset. Right we are only using field<vec> but maybe when we implement segmentation models we might end up needing field<mat> (maybe). For you input will be field<template> only.

Reply:
In that case, isn't this PR good to go?

Reply:
We should one or two more test to test against the new type as well.

Reply:
Sure!